src/NodeParser/TypeReferenceNodeParser.ts

```diff
@@ -20,41 +21,61 @@ export class TypeReferenceNodeParser implements SubNodeParser {
     }
 
     public createType(node: ts.TypeReferenceNode, context: Context): BaseType {
-        const typeSymbol = this.typeChecker.getSymbolAtLocation(node.typeName)!;
+        const typeSymbol =
+            this.typeChecker.getSymbolAtLocation(node.typeName) ??
+            // When the node doesn't have a valid source file, its position is -1, so we can't
+            // search for a symbol based on its location. In that case, the ts.factory defines a symbol
+            // property on the node itself.
+            (node.typeName as unknown as ts.Type).symbol;
+
+        // Wraps promise type to avoid resolving to a empty Object type.
+        if (typeSymbol.name === "Promise") {
+            return this.childNodeParser.createType(node.typeArguments![0]!, this.createSubContext(node, context));
+        }
+
         if (typeSymbol.flags & ts.SymbolFlags.Alias) {
             const aliasedSymbol = this.typeChecker.getAliasedSymbol(typeSymbol);
+
             return this.childNodeParser.createType(
-                aliasedSymbol.declarations!.filter((n: ts.Declaration) => !invalidTypes[n.kind])[0],
+                aliasedSymbol.declarations!.filter((n: ts.Declaration) => !invalidTypes[n.kind])[0]!,
```

Comment:
How do we guarantee that the declarations are not empty? Let's add a comment why the ! is okay here.

Reply:
Actually, it doesn't need to. This code is the same as the current next branch.

I don't remember much, but I probably added these non null assertions because the type checker was emitting some sort of error...